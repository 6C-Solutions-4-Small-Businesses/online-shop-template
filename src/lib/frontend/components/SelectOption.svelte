<script lang="ts">
    export let label: string
    export let isSelected: boolean
    export let onClickHandler: () => void

</script>

<button
        class={`h-14 flex justify-start items-center shadow-md pl-3.5 py-4 hover:bg-orange-100 hover:text-red-500 rounded-md ${isSelected ? 'bg-orange-100 text-red-500' : 'bg-white'}`}
        on:click={onClickHandler}
>
    {label}
</button>