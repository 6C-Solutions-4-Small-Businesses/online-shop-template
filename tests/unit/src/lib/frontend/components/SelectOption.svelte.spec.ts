import {render, type RenderResult} from '@testing-library/svelte'
import SelectOption from '$lib/frontend/components/SelectOption.svelte'
import {expect, vi} from 'vitest'

let rendering: RenderResult<SelectOption, typeof import('@testing-library/dom/types/queries')>


describe('Select option Component', () => {
    describe('html structure', () => {

        beforeEach(() => {
            rendering = render(SelectOption, {
                label: 'Option 1',
                isSelected: false,
                onClickHandler: vi.fn()
            })
        })

        it('should have match snapshot', () => {
            expect(rendering.container).toMatchSnapshot()
        })
    })
})