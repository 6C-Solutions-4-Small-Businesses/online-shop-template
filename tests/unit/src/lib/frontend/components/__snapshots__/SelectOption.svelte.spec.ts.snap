// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Select option Component > html structure > should have match snapshot 1`] = `
<div>
  <button
    class="h-14 flex justify-start items-center shadow-md pl-3.5 py-4 hover:bg-orange-100 hover:text-red-500 rounded-md bg-white"
  >
    Option 1
  </button>
</div>
`;
